specs/HostResourceMapping.md

```diff
@@ -0,0 +1,108 @@
+# Background
+We have received feedbacks that there is a need to access local files in WebView2. In this document we describe the APIs for this scenario. We'd appreciate your feedback.
+
+
+# Description
+To access local files in WebView2, define a virutal host name to local folder mapping using the `AddVirtualHostNameToFolderMapping` API,
+and then refer to local files under the folder with a normal http/https url with the virtual host name.
+
+# Examples
+Suppose the app has index.html file next to its exe file. The following code snippet demonstrates how to define a mapping
+for `app-file.invalid` host name so that it can be accessed from pages using normal http/https url like https://app-file.invalid/index.html.
+
+## Win32 C++
+```cpp
+webview2->AddVirtualHostNameToFolderMapping(
+    L"app-file.invalid", L".", COREWEBVIEW2_HOST_RESOURCE_ACCESS_KIND_DENY_CORS);
```

Comment:
1) Should the hostname be in A-label (punycoded) format, or in U-label format?
2) If you pass an invalid name e.g. "a...b" which has multiple adjacent dots, is the API expected to fail/throw?

I believe the correct answers would be "we just pass it to the HostName constructor inside this API and so inherit whatever it does".  Can you instead just take a HostName instance instead of a string?

Reply:
We can use Windows.Networking.HostName for our WinRT API surface. For our COM and .NET APIs I don't see a corresponding class representing a hostname and so I imagine we'll still want to use string and still need to document the answers to your questions 1 and 2.

Reply:
If I add `APP-FILE.INVALID`, will that match `http://app-file.invalid`? What about IDN? What about trailing dots (`app-file.invalid.` vs. `app-file.invalid`?)

Reply:
Yes, great point. We need to document the normalization applied to the hostname.

Reply:
If you use the existing HostName constructor, you don't need to document the normalization applied, since it's part of an existing API.  You just have to reference it, or just use the type.

Reply:
Right and we will for our WinRT API. For our COM and .NET APIs which can run on Win7 where we won't have that type, we must still use strings and so must document the normalization applied by chromium's hostname parser.

Reply:
Updated with detailed behavior.
Reserves a special domain appassets.webview2.microsoft.com for apps to use.
Also uses HostName for WinRT API. Internal implementation will still get string from HostName and uses Chromium code to validate the input value.

Reply:
After some discussion, we will keep using string as input hostName for WinRT API as well. That would make the APIs consistent among different projections and platform. Internally, we use Chromium's code to canonicalize host name, which might have different logic from HostName.